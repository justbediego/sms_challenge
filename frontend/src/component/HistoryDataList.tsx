import * as React from "react";
import {Datagrid, DateField, List, NumberField, TextField} from 'react-admin';
import "./HistoryDataList.scss"

export const HistoryDataList = (props: any) => (
    <List {...props}>
        <Datagrid rowClick="edit">
            <NumberField source="id"/>
            <TextField source="city"/>
            <DateField source="startDate"/>
            <DateField source="endDate"/>
            <NumberField source="price"/>
            <TextField source="status"/>
            <TextField source="color"/>
        </Datagrid>
    </List>
);