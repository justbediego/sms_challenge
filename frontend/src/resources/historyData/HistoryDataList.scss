.color-box{
  border-radius: 3px;
  border: 1px solid;
  width: 15px;
  height: 15px;
}