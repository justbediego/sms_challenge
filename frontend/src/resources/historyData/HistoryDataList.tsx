import * as React from "react";
import {Datagrid, DateField, DateInput, List, NumberField, SelectField, TextField, TextInput} from 'react-admin';
import "./HistoryDataList.scss"

const {ColorField} = require('react-admin-color-input');

const postFilters = [
    <TextInput label="Keyword" source="keyword" alwaysOn/>,
    <DateInput label="From" source="fromDate" alwaysOn/>,
    <DateInput label="To" source="toDate" alwaysOn/>
];

export const HistoryDataList = (props: any) => (
    <List {...props} filters={postFilters}>
        <Datagrid rowClick="edit">
            <NumberField source="id"/>
            <TextField source="city"/>
            <DateField source="startDate" locales="de-DE"/>
            <DateField source="endDate" locales="de-DE"/>
            <NumberField source="price"/>
            <SelectField source="status" choices={[
                {id: 100, name: 'Once'},
                {id: 101, name: 'Daily'},
                {id: 102, name: 'Weekly'},
                {id: 103, name: 'Yearly'},
                {id: 104, name: 'Seldom'},
                {id: 105, name: 'Monthly'},
                {id: 106, name: 'Often'},
                {id: 107, name: 'Never'}
            ]}/>
            <ColorField source="color"/>
        </Datagrid>
    </List>
);