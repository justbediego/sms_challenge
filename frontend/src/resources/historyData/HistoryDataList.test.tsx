import React from 'react';
import {HistoryDataList} from './HistoryDataList';
import {TestContext} from "ra-test";
import {ThemeProvider} from '@material-ui/styles';
import {createTheme} from '@material-ui/core/styles';
import ReactDOMServer from 'react-dom/server';

describe('HistoryDataList', () => {
    let testUtils: string;

    beforeEach(() => {
        const defaultEditProps = {
            basePath: 'basePath',
            id: 'testID',
            resource: 'testResource',
        };
        const theme = createTheme({});

        testUtils = ReactDOMServer.renderToStaticMarkup(
            <ThemeProvider theme={theme}>
                <TestContext enableReducers={true}>
                    <HistoryDataList {...defaultEditProps} />
                </TestContext>
            </ThemeProvider>
        );
    });

    test('matches snapshot', () => {
        expect(testUtils
            // .replace(/id="mui-[0-9]*"/g, 'id="mui-id"')
            // .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, 'aria-labelledby="mui-area-id"')
        ).toMatchSnapshot();
    });

});