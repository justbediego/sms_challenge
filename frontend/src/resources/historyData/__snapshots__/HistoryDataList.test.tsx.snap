// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryDataList matches snapshot 1`] = `"<div class=\\"list-page RaList-root-1\\" id=\\"testID\\" filter=\\"[object Object]\\"><div class=\\"MuiToolbar-root MuiToolbar-regular RaListToolbar-toolbar-7 MuiToolbar-gutters\\"><form class=\\"RaFilterForm-form-9\\" __versions=\\"[object Object]\\"><div data-source=\\"keyword\\" class=\\"filter-field RaFilterFormInput-body-11\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginDense\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled\\" data-shrink=\\"false\\" for=\\"keyword\\" id=\\"keyword-label\\"><span>Keyword</span></label><div class=\\"MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense\\"><input type=\\"text\\" aria-invalid=\\"false\\" aria-describedby=\\"keyword-helper-text\\" id=\\"keyword\\" name=\\"keyword\\" value=\\"\\" class=\\"MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense\\"/></div><p class=\\"MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-marginDense\\" id=\\"keyword-helper-text\\"></p></div><div class=\\"RaFilterFormInput-spacer-12\\"> </div></div><div data-source=\\"fromDate\\" class=\\"filter-field RaFilterFormInput-body-11\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginDense\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-filled\\" data-shrink=\\"true\\" for=\\"fromDate\\" id=\\"fromDate-label\\"><span>From</span></label><div class=\\"MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense\\"><input type=\\"date\\" aria-invalid=\\"false\\" aria-describedby=\\"fromDate-helper-text\\" id=\\"fromDate\\" name=\\"fromDate\\" value=\\"\\" class=\\"MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense\\"/></div><p class=\\"MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-marginDense\\" id=\\"fromDate-helper-text\\"></p></div><div class=\\"RaFilterFormInput-spacer-12\\"> </div></div><div data-source=\\"toDate\\" class=\\"filter-field RaFilterFormInput-body-11\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginDense\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-filled\\" data-shrink=\\"true\\" for=\\"toDate\\" id=\\"toDate-label\\"><span>To</span></label><div class=\\"MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense\\"><input type=\\"date\\" aria-invalid=\\"false\\" aria-describedby=\\"toDate-helper-text\\" id=\\"toDate\\" name=\\"toDate\\" value=\\"\\" class=\\"MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense\\"/></div><p class=\\"MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-marginDense\\" id=\\"toDate-helper-text\\"></p></div><div class=\\"RaFilterFormInput-spacer-12\\"> </div></div><div class=\\"RaFilterForm-clearFix-10\\"></div></form><span></span><div class=\\"MuiToolbar-root MuiToolbar-regular RaTopToolbar-root-19 RaListToolbar-actions-8 MuiToolbar-gutters\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text RaButton-button-21 MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"ra.action.export\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root RaButton-smallIcon-24\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z\\"></path></svg><span class=\\"RaButton-label-22\\">ra.action.export</span></span></button></div></div><div class=\\"RaList-main-2\\"><div class=\\"MuiPaper-root MuiCard-root RaList-content-3 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiToolbar-root MuiToolbar-regular RaBulkActionsToolbar-toolbar-27 RaBulkActionsToolbar-collapsed-30 MuiToolbar-gutters\\" data-test=\\"bulk-actions-toolbar\\"><div class=\\"RaBulkActionsToolbar-title-31\\"><button class=\\"MuiButtonBase-root MuiIconButton-root RaBulkActionsToolbar-icon-32 MuiIconButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"ra.action.unselect\\" title=\\"ra.action.unselect\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></span></button><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit\\">ra.action.bulk_actions</h6></div><div class=\\"MuiToolbar-root MuiToolbar-regular RaTopToolbar-root-19 RaBulkActionsToolbar-topToolbar-28 MuiToolbar-gutters\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text RaButton-button-21 RaBulkDeleteWithUndoButton-deleteButton-33 MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"ra.action.delete\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root RaButton-smallIcon-24\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg><span class=\\"RaButton-label-22\\">ra.action.delete</span></span></button></div></div><span class=\\"MuiTablePagination-root\\" filter=\\"[object Object]\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters\\"><div class=\\"MuiTablePagination-spacer\\"></div><p class=\\"MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit\\">ra.navigation.page_rows_per_page</p><div class=\\"MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot\\"><div class=\\"MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\">10</div><input value=\\"10\\" aria-hidden=\\"true\\" tabindex=\\"-1\\" class=\\"MuiSelect-nativeInput\\"/><svg class=\\"MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div><p class=\\"MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit\\">ra.navigation.page_range_info</p><div class=\\"RaPaginationActions-actions-50\\"></div></div></span></div></div></div>"`;
